pkg/reporter/sarif_reporter.go

```diff
@@ -0,0 +1,127 @@
+package reporter
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type SARIFReporter struct {
+	outputDest string
+}
+
+type SARIFLog struct {
+	Version string `json:"version"`
+	Schema  string `json:"$schema"`
+	Runs    []runs `json:"runs"`
+}
+
+type runs struct {
+	Tool    tool     `json:"tool"`
+	Results []result `json:"results"`
+}
+
+type tool struct {
+	Driver driver `json:"driver"`
+}
+
+type driver struct {
+	Name    string `json:"name"`
+	InfoURI string `json:"informationUri"`
+	Version string `json:"version"`
+}
+
+type result struct {
+	Kind      string     `json:"kind"`
+	Level     string     `json:"level"`
+	Message   message    `json:"message"`
+	Locations []location `json:"locations"`
+}
+
+type message struct {
+	Text string `json:"text"`
+}
+
+type location struct {
+	PhysicalLocation physicalLocation `json:"physicalLocation"`
+}
+
+type physicalLocation struct {
+	ArtifactLocation artifactLocation `json:"artifactLocation"`
+}
+
+type artifactLocation struct {
+	URI string `json:"uri"`
+}
+
+func NewSARIFReporter(outputDest string) *SARIFReporter {
+	return &SARIFReporter{
+		outputDest: outputDest,
+	}
+}
+
+func createSARIFReport(reports []Report) (*SARIFLog, error) {
+	var log SARIFLog
+
+	n := len(reports)
+
+	log.Version = "2.1.0"
+	log.Schema = "https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/schemas/sarif-schema-2.1.0.json"
+
+	log.Runs = make([]runs, 1)
+	runs := &log.Runs[0]
+
+	runs.Tool.Driver.Name = "config-file-validator"
+	runs.Tool.Driver.InfoURI = "https://github.com/Boeing/config-file-validator"
+	runs.Tool.Driver.Version = "1.7.1"
```

Comment:
This shouldn't be hardcoded as we'll have to remember to change it for every release

Reply:
done

Reply:
@shiina4119 Actually this is still hardcoded. I don't want to have to remember to go into here and update it each release

Reply:
The only way I can see to get `version` is to call the top-level module `configfilevalidator` inside the `reporter` package and calling `configfilevalidator.GetVersion().Version`. Wouldn't this introduce a circular dependency?

```go
package reporter

import (
	"encoding/json"
	"fmt"
	"strings"

	configfilevalidator "github.com/Boeing/config-file-validator"
)

const SARIFVersion = "2.1.0"
const SARIFSchema = "https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/schemas/sarif-schema-2.1.0.json"
const DriverName = "config-file-validator"
const DriverInfoURI = "https://github.com/Boeing/config-file-validator"
var DriverVersion = configfilevalidator.GetVersion().Version
```

Is there any other way to get `version` that I am missing?
Should I go ahead with this?